os/root-filesystem-tricks.md

```diff
@@ -0,0 +1,71 @@
+
+### Overview
+Partitions used in the RAID array containing the root filesystem need to have the type GUID `be9067b9-ea49-4f15-b4f6-f36f8c9e1818` to ensure the initramfs can find the root partition within. In order to have the root partition in a RAID array:
+
+ * Create the partitions used in the RAID array with the GPT partition type GUID `be9067b9-ea49-4f15-b4f6-f36f8c9e1818`
+ * Create a RAID array with the created partitions 
+ * Create a filesystem with on the RAID array with the ROOT label
+ * Remove the ROOT label from the old root filesystem by wiping the existing filesystem or overwriting it
+
+### Example Container Linux Config
+This Container Linux Config creates partitions on /dev/vdb and /dev/vdc consuming the whole disk, then creates a RAID array using the partitions and creates the root filesystem on the array.
+*Warning: This will wipe both /dev/vdb and /dev/vdc.*
+```yaml container-linux-config
+storage:
+  disks:
+    - device: /dev/vdb
+      wipe_table: true
+      partitions:
+       - label: root1
+         number: 1
+         size: 0
+         start: 0
+         type_guid: be9067b9-ea49-4f15-b4f6-f36f8c9e1818
+    - device: /dev/vdc
+      wipe_table: true
+      partitions:
+       - label: root2
+         number: 2
+         size: 0
+         start: 0
+         type_guid: be9067b9-ea49-4f15-b4f6-f36f8c9e1818
+  RAID:
+    - name: "root_array"
+      level: "RAID1"
+      devices:
+        - "/dev/vdb1"
+        - "/dev/vdc1"
+  filesystems:
+    - name: "ROOT"
+      mount:
+        device: "/dev/md/root_array"
+        format: "ext4"
+        create:
+          options:
+            - "-L"
+            - "ROOT"
+    - name: "unused"
+      mount:
+        device: "/dev/vda9"
+        format: "ext4"
+        create:
+          options:
+            - "-L"
+            - "unused"
+          force: true
+```
+
+### Limitations
+
+ * The USR partitions cannot be on a RAID array.
+ * Since Ignition cannot create partitions without first wiping a partition table, the existing root partition cannot be reused in a RAID array.
+ * Since `mdadm.conf` is on the root filesystem, it cannot be read until the root filesystem is mounted and therefore cannot be used to configure a RAID array containing the root filesystem.
+ * Because the initramfs uses the GPT type GUID to find RAID components with the root filesystem, any RAID component with the root filesystem must be partitioned, even if that means having one partition consume the entire disk.
+
+### Future
+Ignition will support resizing and relabling partitions which allow the root partition to be resized and assigned a new type GUID. This will allow it to be reused as part of a RAID array.
```

Comment:
Should this move up be part of the Ignition limitation explanation or just be removed completely?